File: easy/0917_reverse_only_letters/solution.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        sList = list(s)

        left = 0
        right = len(sList) - 1

        while left < right:
            if sList[left].isalpha() and sList[right].isalpha():
                sList[left], sList[right] = sList[right], sList[left]
                left += 1
                right -= 1
            elif sList[left].isalpha():
                right -= 1
            else:
                left += 1

        return "".join(sList)

File: easy/0917_reverse_only_letters/solution.cpp
#include <string>

class Solution {
public:
  std::string reverseOnlyLetters(std::string s) {
    size_t left = 0;
    size_t right = s.size() - 1;

    while (left < right) {
      if (isalpha(s[left]) && isalpha(s[right])) {
        std::swap(s[left], s[right]);
        ++left;
        --right;
      } else if (isalpha(s[left]))
        --right;
      else
        ++left;
    }

    return s;
  }
};
